File: android/lib/src/main/java/kl/open/fmandroid/MatchResult.kt
package kl.open.fmandroid

/**
 * Represents the result of the face capture and match operation.
 *
 * @property processed Indicates whether the processing job completed, independently by the result.
 * @property referenceIsValid Indicates whether the reference image was valid and usable (e.g., face detected).
 * @property capturedIsLive True if the captured face passed liveness detection.
 * @property isSameSubject True if the captured face matches the reference face.
 * @property capturedPath Path to the captured image, or null if the capture failed.
 */
data class MatchResult(
    val processed: Boolean,
    val referenceIsValid: Boolean,
    val capturedIsLive: Boolean,
    val isSameSubject: Boolean,
    val capturedPath: String?
) {
    companion object {
        fun error(): MatchResult {
            return MatchResult(
                false,
                false,
                false,
                false,
                null
            )
        }
    }
}

File: ios/FaceMatchSDK/FaceMatchSDK/models/MatchResult.swift
import Foundation

/// A structure representing the result of the capture and face matching operation.
public struct MatchResult {
    /// Indicates whether the processing job completed, independently by the result.
    public let processed: Bool

    /// Indicates whether the reference image was valid and usable (e.g., face detected).
    public let referenceIsValid: Bool

    /// Indicates whether the captured image passed the liveness check.
    public let capturedIsLive: Bool

    /// Indicates whether the captured face matches the reference face.
    public let isSameSubject: Bool

    /// The file path to the captured image (if available).
    public let capturedPath: String?
}

public func matchResultErr() -> MatchResult {
    return MatchResult(
        processed: false,
        referenceIsValid: false,
        capturedIsLive: false,
        isSameSubject: false,
        capturedPath: nil
    )
}
